tests/auto/plugins/pol/generatecase.h

```diff
@@ -0,0 +1,199 @@
+            data[i] = (gen() % 0x5E) + 0x20;
+        }
+        auto result = pol::convert<char, char32_t>(data, conv);
+        iconv_close(conv);
+        return result;
+    }
+
+    case pol::PolicyRegType::REG_MULTI_SZ: {
+        std::vector<std::string> data1;
+        size_t count = rand() % 100;
+        for (size_t i = 0; i < count; ++i) {
+            std::basic_string<char32_t> data;
+            data.resize((rand() % 100) + 1);
+            for (size_t i = 0; i < data.size(); ++i) {
+                data[i] = (gen() % 0x5E) + 0x20;
+            }
+            data1.push_back(pol::convert<char, char32_t>(data, conv));
+        }
+        iconv_close(conv);
+        return data1;
+    }
+
+    case pol::PolicyRegType::REG_BINARY: {
+        std::vector<uint8_t> data;
+        size_t count = gen() % 100;
+        for (size_t i = 0; i < count; ++i) {
+            data.push_back((gen() % 255) + 1);
+        }
+        iconv_close(conv);
+        return data;
+    }
+
+    case pol::PolicyRegType::REG_DWORD_LITTLE_ENDIAN:
+        iconv_close(conv);
+        return uint32_t(rand() % 10'000'000);
+    case pol::PolicyRegType::REG_DWORD_BIG_ENDIAN:
+        iconv_close(conv);
+        return uint32_t(rand() % 10'000'000);
+    case pol::PolicyRegType::REG_QWORD_LITTLE_ENDIAN:
+        iconv_close(conv);
+        return uint64_t(rand() % 10'000'000);
+    case pol::PolicyRegType::REG_QWORD_BIG_ENDIAN:
+        iconv_close(conv);
+        return uint64_t(rand() % 10'000'000);
+    default:
+        iconv_close(conv);
+        return {};
+    }
+}
+
+void generateCase(size_t last, size_t seed = -1)
+{
+    std::mt19937 gen;
+    std::uniform_int_distribution<std::mt19937::result_type> dist(0, 500);
+    auto parser = pol::createPregParser();
+    size_t current = 0;
+
+    if (seed == -1) {
+        std::random_device dev;
+        seed = dev();
+        gen.seed(seed);
+    } else {
+        gen.seed(seed);
+    }
+
+    std::cout << std::endl << "Begin test with generating cases. Seed: " << seed << std::endl;
+
+    while (current <= last) {
+        std::stringstream file;
+
+        // Generate case
+        pol::PolicyFile data;
+        data.body = std::make_optional<pol::PolicyBody>();
+        size_t el = dist(gen);
+        for (size_t i = 0; i < el; i++) {
+            pol::PolicyInstruction instruction;
+            instruction.type = generateRandomType(gen);
+            instruction.data = generateRandomData(instruction.type, gen);
+            data.body->instructions[generateRandomKeypath(gen)][generateRandomValue(gen)] =
+                    std::move(instruction);
+        }
+
+        parser->write(file, data);
+        file.seekg(0, std::ios::beg);
+
+        auto test = parser->parse(file);
+        if (data != test) {
+            std::cerr << "error: one of generated files detect error in parser." << std::endl;
+            assert(0);
+        }
+
+        std::cout << "Generated " << current << " case: OK" << std::endl;
```

Comment:
Print generated random test case values like key, value, etc. to `stdout`. Maybe here it is better to use `QDebug`.